### PR TITLE
Feat: add dataverse items dynamic routes

### PR DESCRIPTION
**Pull request by lolottetheclash**
The PR implements this [issue](https://github.com/okp4/dataverse-portal/issues/97).
It enables clic on dataverse card buttons located in Home and in Explore pages.
For the moment, if the URL id param doesn't exist, a message "Dataspace/Dataset/Service is not found" appears while waiting for the layout of the 404 page. 
🤔 The data as well as the way of querying these data is going to change, a refactor will be done at this time to decouple the elements.

### CODE REVIEW

File: src/component/card/dataverseCard/dataverseCard.tsx [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
 
 export type DataverseCardProps = {
+  id: string
   type: DataverseItem
   label: string
   description: string
 }
 
-export const DataverseCard: FC<DataverseCardProps> = ({ type, label, description }) => {
+export const DataverseCard: FC<DataverseCardProps> = ({ id, type, label, description }) => {
```
Comment (ccamel) [inline src/component/card/dataverseCard/dataverseCard.tsx:21]: If you are targeting a dataverse item, I'd call it: DataverseItemCard (and the props accordingly).

File: src/component/card/dataverseCard/dataverseCard.tsx [hunk 2]
```diff
@@ -36,6 +39,10 @@ ${description}`,
     []
   )
 
+  const handleDataverseItemDetails = (id: string) => () => {
+    navigate(`/dataverse/${type}/${id}`)
+  }
```
Comment (ccamel) [inline src/component/card/dataverseCard/dataverseCard.tsx:44]: Unexpected closure here. You need to make a decision between two options: passing all values as arguments (id, type) or capturing all values.

File: src/page/dataverse/dataset/dataset.tsx [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { getResourceDetails } from '../dataverse'
+import type { DataverseItemDetails } from '../dataverse'
+
+const Dataset = (): JSX.Element => {
+  const { id } = useParams<string>()
+  const [dataset, setDataset] = useState<DataverseItemDetails | undefined>(undefined)
```
Comment (ccamel) [inline src/page/dataverse/dataset/dataset.tsx:8]: Use implicit argument.

```suggestion
  const [dataset, setDataset] = useState<DataverseItemDetails | undefined>()
```

File: src/page/dataverse/dataspace/dataspace.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { useEffect, useState } from 'react'
+import { useParams } from 'react-router-dom'
+import { getResourceDetails } from '../dataverse'
+import type { DataverseItemDetails } from '../dataverse'
+
+const Dataspace = (): JSX.Element => {
+  const { id } = useParams<string>()
+  const [dataspace, setDataspace] = useState<DataverseItemDetails | undefined>(undefined)
```
Comment (ccamel) [inline src/page/dataverse/dataspace/dataspace.tsx:8]: Use implicit argument.

```suggestion
  const [dataspace, setDataspace] = useState<DataverseItemDetails | undefined>()
```

File: src/page/dataverse/dataverse.tsx [hunk 5]
```diff
@@ -46,125 +46,144 @@ const filters: Filter[] = [
...
   {
+    id: '20',
     type: 'dataset',
     label: 'SIRENE database - agriculture, forestry and fishing',
     description:
       'This dataset refers to all the agricultural, forestry and fishing companies in France in activity. Obtained through a sequence of data processing using the OKP4 protocol.'
   }
 ]
 
+export const getResourceDetails = (id: string): DataverseItemDetails | undefined => {
```
Comment (ccamel) [inline src/page/dataverse/dataverse.tsx:183]: Time to use [Option](https://gcanti.github.io/fp-ts/modules/Option.ts.html) here.

```suggestion
export const getResourceDetails = (id: string): Option<DataverseItemDetails> => {
```

**Review by ccamel — CHANGES_REQUESTED** [review]
A few comments that I would love for you to consider. At your disposal. Thanks. 😌

File: src/component/card/dataverseCard/dataverseCard.tsx (hunk 1 above)
Comment (lolottetheclash) [reply src/component/card/dataverseCard/dataverseCard.tsx:21]: you made a good point.

File: src/page/dataverse/dataset/dataset.tsx (hunk 3 above)
Comment (lolottetheclash) [reply src/page/dataverse/dataset/dataset.tsx:8]: 🫢

File: src/component/card/dataverseCard/dataverseCard.tsx (hunk 2 above)
Comment (lolottetheclash) [reply src/component/card/dataverseCard/dataverseCard.tsx:44]: I missed it, thanks

File: src/page/dataverse/dataverse.tsx (hunk 5 above)
Comment (lolottetheclash) [reply src/page/dataverse/dataverse.tsx]: powerful and complex !

**Review by ccamel — APPROVED** [review]
Great. Very good use of functional programming. I like that. 😊

**Pull request merged**